crates/env/src/api.rs

```diff
@@ -621,6 +621,57 @@ where
 /// # Errors
 ///
 /// `ReturnCode::CodeNotFound` in case the supplied `code_hash` cannot be found on-chain.
+///
+/// # Storage Compatibility
+/// When the smart contract code is modified, it is important to observe an additional restriction
+/// that is imposed on this procedure: you cannot change the order in which the contract state variables
+/// are declared, nor their type.
```

Comment:
Maybe it should be made absolutely clear that it is not a restriction per se, there is nothing to prevent you doing it. It is rather the consequences of having new storage data types which are incompatible with the existing stored data. In the best case it will result in the new contract simply failing to load the data from storage, in the worst case it could read the existing data into the new storage struct incorrectly, leading to unwanted side effects.

Reply:
Nit: Could you add an empty line after the `#`, so that we have it consistent everywhere? Also something like 80 CPL would be great for consistency.

Reply:
> Nit: Could you add an empty line after the `#`, so that we have it consistent everywhere? Also something like 80 CPL would be great for consistency.

Addressed this in the last commit